### ISSUE TITLE
Types for complex.__new__ should be any numeric type

### ISSUE THREAD

**Issue by smlerman** [microsoft/pyright#3763]
The types for the arguments to `complex.__new__` should be any numeric type, not just float.

```
c1 = complex(1, 2)
c2 = complex(1, c1)
```
This code is valid, but the second line reports "complex is incompatible with float".

From https://docs.python.org/3/library/functions.html#complex, "Each argument may be any numeric type (including complex)."

https://github.com/microsoft/pyright/blob/b05dfb2a007a6e087930bbee281088e7125a2ad5/packages/pyright-internal/typeshed-fallback/stdlib/builtins.pyi#L368

**Comment by erictraut** [comment]
Pyright is reporting the correct type error here given the type information provided in the typeshed stubs. The `complex.__new__` method is defined as follows in typeshed's `builtins.pyi` stub:

```python
class complex:
    @overload
    def __new__(cls: type[Self], real: float = ..., imag: float = ...) -> Self: ...
    @overload
    def __new__(cls: type[Self], real: str | SupportsComplex | SupportsIndex | complex) -> Self: ...
```

In your example, you're passing two positional arguments, so it would be covered by the first overload, which indicates that the second positional argument must be a `float`.

If you think this is an error, please file a bug or a PR in the [typeshed project](https://github.com/python/typeshed). Pyright (along with other Python type checkers) consumes the type information from typeshed.

**Comment by smlerman** [comment]
Thanks, I'll file the bug report upstream.
